Wabbajack/View Models/CompilerVM.cs

```diff
@@ -2,19 +2,20 @@
 using ReactiveUI.Fody.Helpers;
 using System;
 using System.IO;
-using System.Linq;
 using System.Reactive.Disposables;
 using System.Reactive.Linq;
 using System.Threading.Tasks;
 using System.Windows.Media.Imaging;
+using DynamicData;
+using DynamicData.Binding;
 using Wabbajack.Common;
 using Wabbajack.Lib;
 
 namespace Wabbajack
 {
     public class CompilerVM : ViewModel
     {
-        public MainWindowVM MWVM { get; }
```

Comment:
This refactor suggests there was a misunderstanding with what this property was for, and how the Queue/Log systems were organized.

WorkQueue is a static singleton.  Logs that get output are also a static singleton, of sorts.  Any view should be able to get at their results and display them if they want, but don't necessarily need their own copy.  MainWindowVM is/was a good spot to put singleton concepts, as we know there's only going to be one, and it's the parent of all.

This MWVM property was so that CompilerVM/InstallerVM have a way of getting to the MainWindowVM to get at those singleton collections.  It's public (and short) so that View's can use them in bindings.  Assuming a datacontext of CompilerVM, `{Binding MWVM.Logs}` would start the bind at CompilerVM, walk up to the MainWindowVM, and bind to the singleton log list for display.  Any subVM could walk up and reuse the same backing data in their displays.

Reply:
Since the `MainWindowVM` now is just the main window which is parent to all other views, I thought it would be better to have elements only in VMs that actually use these:
The only VMs that use Work Queue and the Log are `InstallerVM` and `CompilerVM`, 2 out of like 10?. This is ofc under the assumption that the user doesnt try to compile and install in one WJ session, (havent tried that since we dont have a back button in either Installer or Compiler views)

Reply:
Yeah, you bring up a good point.  I think theres two halves to consider:
1)  The wiring of -how- to construct the list of CPU states and the list of Log events.  Rx is nice because we can construct a very complex observable that is instructions for how to get X, but we don't have to subscribe to it until we actually want that wiring to be set up and start running.
2)  The usage of that wiring to coalesce the results into a list for GUI display.


So your solution proposed in this PR is good for (2), as only the InstallationVM and CompilationVM would have any concepts of Log/Cpu setup at all.  If the user just went to browse, no CPU/Log compilation work will be done.  Only if they open InstallationVM/CompilationVM would that logic exist and take CPU.

Problem is now we're duplicating code in every VM that has interest in CPU/Log, so it's terrible for (1).

A good balance would be to define just the observable in a common place, either a static function, or as a member in a singleton class (MainWindowVM).   Then any interested VMs can reference that observable, subscribe, and coalesce into a list for display.  Best of both worlds.

Some caveats on that as well though.  Now if we do add back/forward buttons, if the user navigates to compiler and installer in the same run (not as rare as you might think for people devving), then we now have to sets of coalesce work being done.  This would get worse as more and more VMs are interested in CPU/Logs.

So one way to avoid this, is to not keep references to InstallerVM/CompilerVM as static Lazy members.   If the user navigates from installer to compiler, a compilerVM will be instantiated and set as the active pane.  The installerVM will be destroyed and cleaned up by the GC.  This means we're good to go again and we only ever have one interested party doing coalesce work.

Downside of that though is related to responsiveness.  Yeah, we're now doing less work overall, technically.. but the "wind up" time is heavier.  Things are not prepped, so when the user clicks between compiler/installer, the installer systems can't show until all the wiring has been constructed again, and all the logs coalesced again.  While probably fine, it starts adding up, and overall makes the app feel sluggish.

So I think it's a balance between doing as little work as you can, while also having most systems prepped and ready to go as quick as possible when the user clicks.  I think what we have already is a good balance.  The work is being done when some parties aren't interested, but at least it's being done only once, and has the upside of being ready immediately when any parties are involved, and the other upside of not having to be reconstructed when transferring between two interested parties.